toolsrc/src/vcpkg/binarycaching.cpp

```diff
@@ -663,6 +663,21 @@ namespace
     const ExpectedS<fs::path>& default_cache_path()
     {
         static auto cachepath = System::get_platform_cache_home().then([](fs::path p) -> ExpectedS<fs::path> {
+            auto maybe_cachepath = System::get_environment_variable("VCPKG_BINARY_CACHE");
+            if (auto p_str = maybe_cachepath.get())
+            {
+                const auto path = fs::u8path(*p_str); //(u8path is deprecated in C++20)
+                const auto status = fs::stdfs::status(path);
+                if (!fs::stdfs::exists(status))
+                    return {"Path to VCPKG_BINARY_CACHE does not exist: " + path.u8string(), expected_right_tag};
+                if (!fs::stdfs::is_directory(status))
+                    return {"Value of environment variable VCPKG_BINARY_CACHE is not a directory: " + path.u8string(),
+                            expected_right_tag};
+                if (!p.is_absolute())
```

Comment:
```suggestion
                if (!path.is_absolute())
```

also this check should go first.